client/app/bundles/course/material/folders/handles.ts

```diff
@@ -1,20 +1,35 @@
+import { defineMessages } from 'react-intl';
+import { AxiosError } from 'axios';
+import { FolderData } from 'types/course/material/folders';
 import { getIdFromUnknown } from 'utilities';
 
 import CourseAPI from 'api/course';
 import { CrumbPath, DataHandle } from 'lib/hooks/router/dynamicNest';
 
+const translations = defineMessages({
+  folderNotFound: {
+    id: 'course.material.folders.FolderShow.folderNotFound',
+    defaultMessage: 'Folder not found',
+  },
+});
+
 const getFolderTitle = async (
   courseUrl: string,
   folderId?: number,
 ): Promise<CrumbPath> =>
   CourseAPI.folders
     .fetch(folderId)
     .then((response) => response.data.breadcrumbs)
+    .catch((error) => {
+      const response = (error as AxiosError<FolderData>).response;
+      if (!response?.data.breadcrumbs) throw new Error('Root folder not found');
```

Comment:
again. If root folder is guaranteed to exist, this line is not needed

Reply:
Handling the edge-case in the event that the root folder was somehow deleted (by console code or through direct database manipulation).

Reply:
got it